react/src/components/ResourceGroupSelect.tsx

```diff
@@ -21,6 +22,7 @@ const ResourceGroupSelect: React.FC<ResourceGroupSelectProps> = ({
   const baiRequestWithPromise = useBaiSignedRequestWithPromise();
   const currentProject = useCurrentProjectValue();
   const [key, checkUpdate] = useUpdatableState('first');
+  const [groupSearch, setGroupSearch] = useState('');
```

Comment:
Can you use `useControllableValue` for this part? Your changes cause the parent componento to lose control over the `search`.